src/dune_engine/fs_memo.mli

```diff
@@ -15,6 +15,9 @@ val file_exists : Path.t -> bool Memo.t
     it. *)
 val dir_exists : Path.t -> bool Memo.t
 
+val dir_exists_with_error :
```

Comment:
This doesn't seem like the right name. Calling this function on a path that doesn't exist will return an error rather than `false` as one might expect. I think this function should be called `is_directory` instead.

Reply:
Fixed.